tests/gui/test_widgets.py

```diff
@@ -2373,6 +2373,45 @@ def test_ReplyBoxWidget_disable(mocker):
     rb.text_edit.set_logged_out.assert_called_once_with()
     rb.send_button.hide.assert_called_once_with()
 
+def test_ReplyTextEdit_focus_change_no_text(mocker):
+    """
+    Tests if placeholder text in reply box disappears when it's focused (clicked)
+    and reappears when it's no longer on focus
+    """
+    source = mocker.MagicMock()
+    controller = mocker.MagicMock()
+    rt = ReplyTextEdit(source, controller)
+    rt.placeholder = mocker.MagicMock()
+
+    rt.setFocus()
+    assert rt.placeholder.hide.assert_called_once_with()
+    assert rt.text() == ''
+
+    rt.clearFocus()
+    assert rt.placeholder.show.assert_called_once_with()
+    assert rt.text() == ''
```

Comment:
I'm having trouble making which simulate the focusing of a widget. It seem that `rt.setFocus()` is not affecting the widget focus state.

Reply:
Just speculating here but maybe this has something to do with the fact that the tests don't really create windows and so Qt can't really set windows with focus or without focus. I also tried replacing the commands with similar ones (`setFocusProxy`, and focus the current window) but it kept failing.

Reply:
hey deeplow, i'll take a look to see if there's another way we can trigger a focus or mouse click event to see that the qlabel hides.

Reply:
so looks like you can get the show/hid check to pass if you create a `QFocusEvent` like this:
```py
    source = mocker.MagicMock()
    controller = mocker.MagicMock()
    rt = ReplyTextEdit(source, controller)

    event = QFocusEvent(QEvent.FocusIn)

    rt.focusInEvent(event)
    assert rt.placeholder.isHidden()
```

Reply:
and you can do the same for FocusOut instead of using `rt.clearFocus()`

also, you probably just haven't gotten to it yet, but `rt.text() == ''` should be updated to `assert rt.toPlainText() == ''` and it'll pass

Reply:
Great. Thanks for looking into this @creviera. I'll add this fix shortly

Reply:
It seems like we're on a good path but `assert rt.placeholder.isHidden()` won't work because `rt.placeholder` is a mock so calling the function `hide()` of the  mock won't make changes to it.

```python
        # change focus out of rt
        focus_out_event = QFocusEvent(QEvent.FocusOut)
        rt.focusOutEvent(focus_out_event)
>       assert rt.placeholder.isHidden() == False
E       AssertionError: assert <MagicMock name='mock.isHidden()' id='128204402006280'> == False
E        +  where <MagicMock name='mock.isHidden()' id='128204402006280'> = <MagicMock name='mock.isHidden' id='128204402002072'>()
E        +    where <MagicMock name='mock.isHidden' id='128204402002072'> = <MagicMock id='128204401943048'>.isHidden
E        +      where <MagicMock id='128204401943048'> = <securedrop_client.gui.widgets.ReplyTextEdit object at 0x7499e9d2f4c8>.placeholder
```

This is why `assert rt.placeholder.isHidden()` passes - it returns an object that is different from False.

I'll dig in a bit

Reply:
In my code, `rt.placeholder` is not a mockup, see:

```py
    source = mocker.MagicMock()
    controller = mocker.MagicMock()
    rt = ReplyTextEdit(source, controller)

    event = QFocusEvent(QEvent.FocusIn)

    rt.focusInEvent(event)
    assert rt.placeholder.isHidden()
```

^ This works for me